src/config.rs

```diff
@@ -121,14 +93,77 @@ impl Default for Templates {
 }
 
 
+pub struct Provider {
+    pub client_id: String,
+    pub secret: String,
+    pub discovery_url: String,
+    pub issuer_domain: String,
+}
+
+
+pub struct Config {
```

Comment:
Renaming `AppConfig` to `Config` should be a commit on its own, too, though I'm not sure what the goal of this is?